beta-src/src/state/game/game-api-slice.ts

```diff
@@ -349,17 +308,75 @@ const drawBuilds = (state) => {
   }
 };
 
+// Destroy a Unit during Builds Phase
+const destroyUnit = (state, unitID) => {
+  const {
+    data: {
+      data: { currentOrders },
+    },
+    order,
+    ordersMeta,
+  }: {
+    data: GameDataResponse;
+    order: OrderState;
+    ordersMeta: OrdersMeta;
+  } = current(state);
+
+  // Check to make sure you have units to destroy
+  const count = currentOrders?.filter(
+    (item) => item.unitID === "" || item.unitID === null,
+  );
+
+  if (count && count?.length === 0 && unitID !== order.unitID) {
+    return;
+  }
+
+  const ordersMetaEntries = Object.entries(ordersMeta);
+
+  Object.values(ordersMeta).forEach(({ update }) => {
+    if (update) {
+      const { type } = update;
+      if (type === "Destroy") {
+        const command: GameCommand = {
+          command: order.inProgress ? "NONE" : "DESTROY",
+          data: {
+            setUnit: {
+              componentType: "Icon",
+              iconState: order.inProgress ? UIState.NONE : UIState.DESTROY,
+              unitSlotName: "main",
+            },
+          },
+        };
+        setCommand(state, command, "unitCommands", unitID);
+
+        state.data.data.currentOrders.forEach((item) => {
+          ordersMetaEntries.forEach((meta) => {
```

Comment:
Im wondering if you even need to use another loop inside this loop. since ordersMeta is naturally an object that is keyed with the order id, cant you just loop through currentOrders and grab its id and check if it exists in ordersMeta like: ordersMeta[item.id]? This should get you the ordersMeta if it exists for that order